### ISSUE TITLE
config file not used with stdin

### ISSUE THREAD

**Issue by darrinholst** [prettier/prettier#2689]
Vim users that are using neoformat for prettier integration use the `--stdin` option which causes the config not to be picked up.

When using stdin `filePath` is the working directory [here](https://github.com/prettier/prettier/blob/master/src/resolve-config.js#L12) making `fileDir` one level up from where it needs to be.

**Comment by vjeux** [comment]
cc @azz: what do you think?

**Comment by mitermayer** [comment]
One other point to consider is that plugins that leverage from --stdin usually would have a hard time to read all types of configurations that prettier support.

Maybe this could be solved by having a CLI flag like `--prefer-config` where in case a configuration file exist would then ignore all other params passed by the CLI, this way plugins can still pass the normal arguments for projects that do not have configuration files.

@vjeux  what are your thoughts around it ?

**Comment by lydell** [comment]
IMO, the config files should be used as usual with stdin.

**Comment by azz** [comment]
Dirty hack like this would fix it:

```patch
-    getOptionsForFile(process.cwd()).then(options => {
+    getOptionsForFile(path.join(process.cwd(), "(stdin)").then(options => {
```

https://github.com/prettier/prettier/blob/master/bin/prettier.js#L327

Otherwise we'd need to check if `filePath` is a file or a directory in resolve-config.js.

**Comment by azz** [comment]
Do'h, that would do it, too. Forgot that cosmiconfig handles files perfectly well.

**Comment by vjeux** [comment]
I just shipped 1.6.1 that contains this fix! Thanks for reporting it!
